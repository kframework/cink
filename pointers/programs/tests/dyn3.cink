#include <iostream>
using namespace std;

int main() {
  int a = 10;
  bool b = true;
  int ** p = new int *;
  bool ** q = new bool *;
  *p = &a;
  *q = &b;
  if (**q) cout << **p << endl;
}

