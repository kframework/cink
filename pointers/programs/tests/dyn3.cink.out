10
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            noVal
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symLoc(1)
    </genv>
    <store>
        #symLoc(1) |-> lambda ( .Decls @ (((((((((int a) ;) ((a = 10) ;)) 
          (((bool b) ;) ((b = true) ;))) ((p of (pointer to (pointer to int)) 
          ;) ((p = (new (int *))) ;))) ((q of (pointer to (pointer to bool)) ;
          ) ((q = (new (bool *))) ;))) (((* p) = (& a)) ;)) (((* q) = (& b)) ;
          )) (if ( (* (* q)) ) (((cout << (* (* p))) << endl) ;) else { })) )
        #symLoc(2) |-> 10
        #symLoc(3) |-> true
        #symLoc(4) |-> #symLoc(5)
        #symLoc(5) |-> #symLoc(2)
        #symLoc(6) |-> #symLoc(7)
        #symLoc(7) |-> #symLoc(3)
    </store>
</T>
